docs/DeveloperGuide.md

```diff
@@ -1,34 +1,173 @@
+    - Set up JDK 11 for Gradle
+        1. Click `Configure` -> `Project Defaults` -> `Project Structure`
+        2. Click `New...` and set it to the directory of the JDK 11 installed
+4. Import the project as a Gradle project.
+    > <font color=#FF5733>Note: </font> <font color=#00000> It is different from importing a normal Java project.</font>
 
-{Give non-functional requirements}
+    If you are using IntelliJ IDEA:
+    - Click `Import Project` (or `Open or Import` in newer version of Intellij).
+    - Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+    - Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK being used for the project.
+    - Wait for importing process to complete
+5. Do the testing. Please follow the testing guide
+   
+## 3. Design 
+### 3.1 Architecture
+The ***Architecture diagram*** given below shows the major components, and explains the structure of the software system.
+[![Architecture diagram](https://iili.io/3Ei2Vf.md.png)](https://freeimage.host/i/3Ei2Vf)
 
-## Glossary
+### 3.2 Classes
 
-* *glossary item* - Definition
+The CEGMods consists of six classes:
+* `Main`
+* `Ui`
+* `Parser`
+* `Module`
+* `Task`
+* `TimeTable`
+
+#### 3.2.1 Ui class
+The Ui class reads the uses's input and response relevant messages.
+#### 3.2.2 Parser class
+The `Parser` class deals with making sense of the user's command and execute the command relevantly:
+- add new `Module` into `Timetable`.
+- delte `Module` from `Timetable`.
+- add new `Task` into tasklist.
+- delete `Task` from tasklist.
+- print out `Timetable` of a certain day or one week.
+  
+#### 3.2.3 Module class
+The Module class contains the information of a module, including the title, description, and time slots of lectures, tutorials and labs.
+#### 3.2.4 Task class
+The `Task` class contains the information of a task, including the todo, event and deadline tasks.
+
+#### 3.2.5 TimeTable class
+The `Timetable` class shows the timetable for all the module slots and tasks. It also stores `Module` details in an arraylist and checks whether there is a time conflict between two classes.
+       
+
+## 4. Implementation
+This section provides details on the implementations of certain features.
+    
+### Feature: Add a module
+
+The user type in command in `module mod/<MODULE_CODE> lec/<LECTURE_DAY> <LECTURE_TIME> tut/<TUTORIAL_DAY> <TUTORIAL_TIME> lab/<LAB_DAY> <LAB_TIME>` this form to add a module into the timetable. The time slot of lab is optional. The program will detect whether the user command is in this form, and store module data in arraylist `modules` in `Timetable` class after Ui passing command and calling `addModule()` in Parser class.
+
+The ***sequence diagram*** below shows the interaction between these classes when the user adds a module.
+[![3GRaM7.md.png](https://iili.io/3GRaM7.md.png)](https://freeimage.host/i/3GRaM7)
```

Comment:
I think the diagram is pretty simple. It helps to show step by step how a module is added. So far no violation of any UML notations.